### DISCUSSION TITLE
Feature request: add an onInit to InstantSearch component (React)

### DISCUSSION THREAD

**Discussion by MidasVE** [algolia/instantsearch#5851] [Ideas]
Issue I currently have:
There is a layout shift when having a lot inside your InstantSearch component because everything in the InstantSearch component gets loaded after the InstantSearch component is loaded.
It would be nice to have an `onInit` or `initialState` or `loadingState` attribute to make it possible to show some kind of a loading state first, then replace the InstantSearch children once initialized.

**Comment by Haroenv** [comment]
Have you considered using server side rendering to prevent the layout shift?
